src/main/java/org/breedinginsight/brapps/importer/services/FileImportService.java

```diff
@@ -429,18 +420,12 @@ public ImportUpload setDynamicColumns(ImportUpload newUpload, Table data, Import
         return newUpload;
     }
 
-    private void processFile(UUID workflowId, List<BrAPIImport> finalBrAPIImportList, Table data, Program program,
+    private void processFile(String workflow, List<BrAPIImport> finalBrAPIImportList, Table data, Program program,
                              ImportUpload upload, User user, Boolean commit, BrAPIImportService importService,
                              AuthenticatedUser actingUser) {
         // Spin off new process for processing the file
         CompletableFuture.supplyAsync(() -> {
             try {
-                Workflow workflow = null;
-                if (workflowId != null) {
-                    Optional<Workflow> optionalWorkflow = workflowFactory.getWorkflow(workflowId);
-                    workflow = optionalWorkflow.orElse(null);
-                }
-
                 ImportServiceContext context = ImportServiceContext.builder()
                         .workflow(workflow)
                         .brAPIImports(finalBrAPIImportList)
```

Comment:
A workflow isn't needed at the FileImportService level, which is why, after instantiating it here with the factory, the workflow had to be passed in by reference to where it is needed at the ExperimentImportService level. This is not desirable as client code receiving the ref has no guarantee that workflow has not been mutated. 

A less error-prone approach is to pass by value any workflow context and instantiate the workflow at the level where it is used, in this case the ExperimentImportService. It is an experiment import workflow instantiated within the ExperimentImportService; I'm not really seeing anything opaque about that.